mozilla-release/browser/app/Makefile.in

```diff
@@ -110,4 +110,4 @@ endif
 
 .PHONY: features
 tools features::
-	$(PYTHON) -c 'import os, json; listing = {"system": os.listdir("$(DIST)/bin/browser/features")}; print json.dumps(listing)' > $(DIST)/bin/browser/chrome/browser/content/browser/built_in_addons.json
+	$(PYTHON) -c 'import os, json; listing = {"system": [os.path.splitext(it)[0] for it in os.listdir("$(DIST)/bin/browser/features")]}; print json.dumps(listing)' > $(DIST)/bin/browser/chrome/browser/content/browser/built_in_addons.json
```

Comment:
can you please add comment on why it is here?
is it because we put unpacked xpi? 
could we put a packed one?

Reply:
This is because in Mozilla people don't use a stuff which they create :) This script generate a new built_in_addons.json file where must be counted all SystemAddons extensions. When they put a file with .xpi extension in the end - later a code looking for absolutely the same id in manifest file. For example, if filename of extension cliqz@cliqz.com.xpi - on launch time Firefox expecting to find the same id in em:id node in install.rdf file.

Reply:
should we ship it upstream then?